### PR TITLE
Feature/download from more than two fg

### PR DESCRIPTION
**Pull request by FelixRottler**
- introduces download from more than one gateway
- this pr adds two new profiles "connect-efgs" and "connect-chgs" that run the download in different configurations. Depending on the profile a discriminator column keeps tracks from which target system the federation batch info comes from

### CODE REVIEW

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/FederationBatchTarget.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package app.coronawarn.server.common.persistence.domain;
+
+public enum FederationBatchTarget {
+  EFGS,
```
Comment (EugenM-SAP) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/FederationBatchTarget.java:4]: Maybe worth to have the unabbreviated explanations on top of each enum

File: services/download/src/main/resources/application.yaml [hunk 2]
```diff
@@ -14,12 +14,13 @@ logging:
 
 services:
   download:
+    target-system: EFGS
```
Comment (EugenM-SAP) [inline services/download/src/main/resources/application.yaml:17]: I would think that having default target system is not ideal. I think we should enforce that a specific connect-xyz profile is always specified and fail fasst otherwise. What do you think?

File: services/download/src/main/resources/bootstrap-cloud.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ spring:
   cloud:
     vault:
       ssl:
-        trust-store: file:${SSL_VAULT_TRUSTSTORE_PATH}
```
Comment (EugenM-SAP) [inline services/download/src/main/resources/bootstrap-cloud.yaml:8]: Are we sure the truststore  is not injected in the container file system?

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/FederationBatchInfo.java [hunk 4]
```diff
@@ -16,27 +16,33 @@
...
+   * @param status   status stored as {@link FederationBatchStatus}
    */
   @PersistenceConstructor
-  public FederationBatchInfo(String batchTag, LocalDate date, FederationBatchStatus status) {
+  public FederationBatchInfo(String batchTag, LocalDate date, FederationBatchStatus status,
+      FederationBatchTarget targetSystem) {
     this.batchTag = batchTag;
     this.date = date;
     this.status = status;
+    this.targetSystem = targetSystem;
```
Comment (EugenM-SAP) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/FederationBatchInfo.java:45]: I know we referred to the gateway as a target system during discussions, but in the case of Download it represents rather a "source" system

File: services/upload/src/main/resources/bootstrap-cloud.yaml [hunk 5]
```diff
@@ -5,7 +5,7 @@ spring:
   cloud:
     vault:
       ssl:
-        trust-store: file:${SSL_VAULT_TRUSTSTORE_PATH}
```
Comment (EugenM-SAP) [inline services/upload/src/main/resources/bootstrap-cloud.yaml:8]: same as question above

File: common/persistence/src/main/resources/db/migration/V8__alterFederationBatchInfoAddSource.sql [hunk 6]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE federation_batch_info
+  ADD source_system VARCHAR(10) NOT NULL;
```
Comment (hilmarf) [inline common/persistence/src/main/resources/db/migration/V8__alterFederationBatchInfoAddSource.sql:2]: I doubt, that this works. Adding new column with `NOT NULL` but without default!?!

File: services/download/src/main/resources/application.yaml (hunk 2 above)
Comment (FelixRottler) [reply services/download/src/main/resources/application.yaml]: ok

File: services/download/src/main/resources/bootstrap-cloud.yaml (hunk 3 above)
Comment (FelixRottler) [reply services/download/src/main/resources/bootstrap-cloud.yaml:8]: I did this because vault ssl was not working on dev when using the prefix "file:". Maybe we need to investigate this

File: common/persistence/src/main/resources/db/migration/V8__alterFederationBatchInfoAddSource.sql (hunk 6 above)
Comment (FelixRottler) [reply common/persistence/src/main/resources/db/migration/V8__alterFederationBatchInfoAddSource.sql:2]: True. Will change it

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/FederationBatchSourceSystem.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package app.coronawarn.server.common.persistence.domain;
+
+/**
+ * Source for downloading  federation batches from different sources.
+ */
+public enum FederationBatchSourceSystem {
+  // EFGS = European Federation Gateway Service
+  EFGS,
+  // CHGS = Swiss Gateway Service
```
Comment (hilmarf) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/FederationBatchSourceSystem.java:9]: why no real javadoc?

**Review by hilmarf — APPROVED** [review]

**Review by ioangut — COMMENTED** [review]
runconfigs/IntelliJ/callback [spring-boot_run...].run.xml, do we need to add these changes into the callback run config? Please remove this file from this PR if not needed.

File: services/callback/src/main/java/app/coronawarn/server/services/callback/controller/CallbackController.java [hunk 8]
```diff
@@ -44,7 +46,8 @@ public CallbackController(FederationBatchInfoService federationBatchInfoService)
   public ResponseEntity<Void> handleCallback(@RequestParam String batchTag,
       @NotNull @DateTimeFormat(iso = ISO.DATE) @RequestParam LocalDate date) {
     logger.info("BatchInfo with tag {} and date {} received from federation gateway.", batchTag, date);
-    FederationBatchInfo federationBatchInfo = new FederationBatchInfo(batchTag, date);
+    // TODO check how to integrate swiss?
```
Comment (ioangut) [inline services/callback/src/main/java/app/coronawarn/server/services/callback/controller/CallbackController.java:49]: Would this TODO be developed in another PR ?

File: services/download/src/main/java/app/coronawarn/server/services/download/Application.java [hunk 9]
```diff
@@ -34,6 +34,8 @@ public static void main(String[] args) {
     SpringApplication.run(Application.class);
   }
 
+
+
```
Comment (ioangut) [inline services/download/src/main/java/app/coronawarn/server/services/download/Application.java:38]: Please remove these empty lines.

**Review by ioangut — APPROVED** [review]

**Review by EugenM-SAP — APPROVED** [review]

File: services/callback/src/main/java/app/coronawarn/server/services/callback/controller/CallbackController.java (hunk 8 above)
Comment (FelixRottler) [reply services/callback/src/main/java/app/coronawarn/server/services/callback/controller/CallbackController.java:49]: yes @EugenM-SAP
